### ISSUE TITLE
Invalid parsing of some URL parts when original URL contains uppercase letters

### ISSUE THREAD

**Issue by satano** [tmenier/Flurl#574]
## Bug description

When original (or source) URL string contains uppercase letters, it is not parsed correctly. This is because of `OriginalString.OrdinalStartsWith(...)` is case sensitive, but `OriginalString` contains uppercase letters as provided on input, but when parsed by `Uri` class, some parts are converted to lowercase.

## Example

``` cs
var lowercaseUrl = new Url(new Uri("http://www.mysite.com:8080"));
Assert.AreEqual("http://www.mysite.com:8080", lowercaseUrl.ToString()); // OK

var uppercaseUrl = new Url(new Uri("http://www.mySite.com:8080")); // Note capital "S".
Assert.AreEqual("http://www.mysite.com:8080", uppercaseUrl.ToString()); // Fails – produces "http://www.mysite.com"
```

## Aditional info

The parsing has changed in stable version Flurl 3.0.0. We have been using pre-release version 3.0.0-pre3 for some time and everything is OK. And we are not able to upgrade to stable version with this bug.

I believe, this is a duplicate of #570. We also found this bug when the port was removed from some of ours URLs, but we have those URLs with uppercase letters. It is because the port is parsed using `_originalString`, so the correct minimal example in issue #570 could be:

``` cs
var requestUri = "https://LOCALHOST:3333".AppendPathSegment("item")
```

**Comment by tmenier** [comment]
Good catch. I've generally adopted the philosophy that the parsed bits should match the original as closely as possible (don't default `Port` if there isn't one, for example), so ideally I'd rather if scheme and host preserved the original casing, but that definitely complicates parsing (I can't lean on `System.Uri` as much), and this is obviously better than the blatant bug that exists now, so I'll go with your PR for now. Thanks!

**Comment by dusanjel** [comment]
@tmenier Hi, when can we expect for this changes to be published in official version?

**Comment by tmenier** [comment]
Fixed & released in 3.0.1
